File: c++/MatrixSort.cpp
#include <iostream> 
using namespace std; 

void sort(int mat[3][3], int row, int col) { 
	int size = row * col; 
	for (int i = 0; i < size; i++) { 
		for (int j = 0; j < size - 1; j++) { 
			if (mat[j / col][j % col] > mat[(j + 1) / col][(j + 1) % col]) { 
				int temp = mat[j / col][j % col]; 
				mat[j / col][j % col] = mat[(j + 1) / col][(j + 1) % col]; 
				mat[(j + 1) / col][(j + 1) % col] = temp; 
			} 
		} 
	} 
} 

void print(int mat[3][3], int row, int col) {

	for (int i = 0; i < row; i++) { 
		for (int j = 0; j < col; j++) 		 
			cout<<mat[i][j];		
		cout << endl; 
	} 
} 

int main() { 
	int mat[3][3] = { { 8, 6, 7 }, 
						{ 2, 4, 5 }, 
						{ 1, 3, 9 } }; 
                        
	sort(mat, 3, 3); 
	print(mat, 3, 3); 
	return 0; 
} 

File: python/MatrixSort.py
def sort(data, row, col): 	
	n = row * col 
	for i in range(0, n): 
		for j in range(0, n-1): 
			
			if ( data[j//col][j % col] >\ data[(j + 1)//col][(j + 1)% col] ):				
				temp = data[j//col][j % col] 
				data[j//col][j % col] =\ data[(j + 1)//col][(j + 1)% col] 
				data[(j + 1)//col][(j + 1)% col] =\ temp 

def print(mat, row, col): 	
	for i in range(row): 
		for j in range(col): 
			print(mat[i][j]) 
		print() 


mat = [ [5, 4, 7], 
			[1, 3, 8], 
			[2, 9, 6] ] 
row = len(mat) 
col = len(mat[0])  
sort(mat, row, col) 
	 
print(mat, row, col) 
